lock-api-objects/src/main/java/com/palantir/lock/v2/TimelockService.java

```diff
@@ -110,4 +112,28 @@ default List<StartIdentifiedAtlasDbTransactionResponse> startIdentifiedAtlasDbTr
     void tryUnlock(Set<LockToken> tokens);
 
     long currentTimeMillis();
+
+    /**
+     * Acquires a lease on a named timestamp. The lease is taken out with a new fresh timestamp.
+     * The returned timestamps are fresh timestamps obtained strictly after the lease is taken out.
+     */
+    @RestrictedApi(
+            explanation =
+                    "This method is for internal Atlas and internal library use only. Clients MUST NOT use it unless"
+                        + " given explicit approval. Mis-use can result in SEVERE DATA CORRUPTION and the API contract"
+                        + " is subject to change at any time.",
```

Comment:
I can add that misuse breaks the protocol but I think they should already be second guessing themselves just from this.